File: src/org/cappuccino_project/ide/intellij/plugin/stubs/interfaces/ObjJClassDeclarationStub.java
package org.cappuccino_project.ide.intellij.plugin.stubs.interfaces;

import com.intellij.psi.stubs.StubElement;
import org.cappuccino_project.ide.intellij.plugin.psi.interfaces.ObjJClassDeclarationElement;
import org.jetbrains.annotations.NotNull;

import java.util.List;

public interface ObjJClassDeclarationStub<PsiT extends ObjJClassDeclarationElement<? extends ObjJClassDeclarationStub>> extends StubElement<PsiT>, ObjJResolveableStub<PsiT> {

    @NotNull
    String getClassName();
    @NotNull
    List<String> getInheritedProtocols();

}

File: src/org/cappuccino_project/ide/intellij/plugin/stubs/interfaces/ObjJClassDeclarationStub.kt
package org.cappuccino_project.ide.intellij.plugin.stubs.interfaces

import com.intellij.psi.stubs.StubElement
import org.cappuccino_project.ide.intellij.plugin.psi.impl.ObjJImplementationDeclarationImpl
import org.cappuccino_project.ide.intellij.plugin.psi.interfaces.ObjJClassDeclarationElement

interface ObjJClassDeclarationStub<PsiT : ObjJClassDeclarationElement<ObjJClassDeclarationStub<PsiT>>> : StubElement<PsiT>, ObjJResolveableStub<PsiT> {

    val className: String
    val inheritedProtocols: List<String>

}